File: integer-to-english-words/integer-to-english-words.py
#Time Complexity : O(n) 
#Space Complexity : O(1) 
​
class Solution:
    def numberToWords(self, num: int) -> str:
​
        below20 = ["", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten" , "Eleven", "Twelve", "Thirteen", "Fourteen", "Fifteen", "Sixteen", "Seventeen", "Eighteen", "Nineteen", "Twenty"]
        tens = ["", "Ten", "Twenty", "Thirty", "Forty", "Fifty", "Sixty", "Seventy", "Eighty", "Ninety"]
        thousands = ["", "Thousand", "Million", "Billion"]
        result = ""
        if num == 0:
            return "Zero"
        def helper(num):
            if num == 0:
                return ""
            elif num<20:
                return below20[num] + " "
            elif num<100:
                return tens[num//10] + " " + helper(num%10)
            else:
                return below20[num//100] + " Hundred " + helper(num%100)
        i=0   
        while num>0:
            if num%1000!=0:
                result = helper(num%1000) + thousands[i] +" "+ result 
            num = num//1000 
            i+=1
​
        return result.strip()

File: integer-to-english-words/integer-to-english-words.cs
            {17, "Seventeen"},
            {16, "Sixteen"},
            {15, "Fifteen"},
            {14, "Fourteen"},
            {13, "Thirteen"},            
            {12, "Twelve"},
            {11, "Eleven"},
            {10, "Ten"},
            {9, "Nine"},
            {8, "Eight"},
            {7, "Seven"},
            {6, "Six"},
            {5, "Five"},
            {4, "Four"},
            {3, "Three"},
            {2, "Two"},
            {1, "One"},
            {0, "Zero"}
        };
    }
​
​
    public string NumberToWords(int num) {
        
foreach(var pair in trials)
{
    if(num <=10 && num==pair.Key)
    {
        return pair.Value;
    }
    
    int n = num / pair.Key;
    num = num % pair.Key;
    if(n >=1)
    {
        if(num==0)
        {
            return String.Format(pair.Value,NumberToWords(n),"").Trim();
        }
        else
        {
            return String.Format(pair.Value,NumberToWords(n),NumberToWords(num));
        }
    }
}
        return "";
        
        
        
        
        
        
    }
}
